### ISSUE TITLE
I feel like Qwen Fine tuning using more vram than what it can do

### ISSUE THREAD

**Issue by FurkanGozukara** [kohya-ss/musubi-tuner#588]
When training FLUX fine tuning there isnt much VRAM usage difference between LoRA vs full fine tuning

Now I am testing same with Qwen Image and there is huge difference

even with 58 block swap, 96 GB RAM and 24 GB VRAM i get OOM

```
additional_parameters = ""
async_upload = false
base_weights_multiplier = 1
blocks_to_swap = 58
caption_strategy = "folder_name"
convert_to_diffusers = false
convert_to_safetensors = false
create_missing_captions = true
custom_network_module = ""
dataset_batch_size = 1
dataset_bucket_no_upscale = false
dataset_cache_directory = "cache_dir"
dataset_caption_extension = ".txt"
dataset_config = "E:\\SECourses_Musubi_Trainer_v8\\test_training\\dataset_config_20250904_033137.toml"
dataset_config_mode = "Generate from Folder Structure"
dataset_control_directory = "edit_images"
dataset_enable_bucket = false
dataset_qwen_image_edit_control_resolution_height = 0
dataset_qwen_image_edit_control_resolution_width = 0
dataset_qwen_image_edit_no_resize_control = false
dataset_resolution_height = 1328
dataset_resolution_width = 1328
ddp_gradient_as_bucket_view = false
ddp_static_graph = false
ddp_timeout = 0
diffusers_output_dir = ""
dim_from_weights = false
disable_prompt_enhancement = true
discrete_flow_shift = 2.2
dit = "E:/SECourses_Musubi_Trainer_v8/Qwen_Image_Training_Models/qwen_image_bf16.safetensors"
dit_dtype = "bfloat16"
dit_in_channels = 16
edit = false
flash3 = false
flash_attn = false
flow_shift = 7
fp8_base = false
fp8_scaled = false
fp8_vl = false
full_bf16 = false
full_fp16 = false
fused_backward_pass = true
generated_toml_path = "E:\\SECourses_Musubi_Trainer_v8\\test_training\\dataset_config_20250904_033137.toml"
gradient_accumulation_steps = 1
gradient_checkpointing = true
gradient_checkpointing_cpu_offload = true
guidance_scale = 4
img_in_txt_in_offloading = false
learning_rate = 5e-5
log_config = false
log_prefix = ""
log_tracker_name = ""
logging_dir = ""
logit_mean = 0
logit_std = 1
lr_decay_steps = 0
lr_scheduler = "constant"
lr_scheduler_args = []
lr_scheduler_min_lr_ratio = 0
lr_scheduler_num_cycles = 1
lr_scheduler_power = 1
lr_scheduler_timescale = 0
lr_scheduler_type = ""
lr_warmup_steps = 0
max_data_loader_n_workers = 2
max_grad_norm = 0
max_timestep = 1000
max_train_epochs = 200
max_train_steps = 90000
mem_eff_save = false
metadata_author = ""
metadata_description = ""
metadata_license = ""
metadata_tags = ""
mixed_precision = "bf16"
mode_scale = 1
network_module = ""
no_metadata = false
optimizer_args = [ "scale_parameter=False", "relative_step=False", "warmup_init=False", "weight_decay=0.01",]
optimizer_type = "AdaFactor"
output_dir = "E:\\SECourses_Musubi_Trainer_v8\\test_training"
output_name = "Tier1_29300_MB_VRAM_Need_More_RAM"
parent_folder_path = "E:\\SECourses_Musubi_Trainer_v8\\test_training\\28_imgs_1328"
persistent_data_loader_workers = true
preserve_distribution_shape = false
safetensors_output_dir = ""
sage_attn = false
sample_at_first = false
sample_cfg_scale = 4
sample_discrete_flow_shift = 2.2
sample_guidance_scale = 1
sample_height = 1328
sample_negative_prompt = ""
sample_seed = 99
sample_steps = 20
sample_width = 1328
save_every_n_epochs = 25
save_state = false
save_state_on_train_end = false
save_state_to_huggingface = false
scale_weight_norms = 0
sdpa = true
seed = 99
show_timesteps = ""
sigmoid_scale = 1
split_attn = true
text_encoder = "E:/SECourses_Musubi_Trainer_v8/Qwen_Image_Training_Models/qwen_2.5_vl_7b_bf16.safetensors"
text_encoder_dtype = "bfloat16"
timestep_sampling = "sigmoid"
training_comment = ""
vae = "E:/SECourses_Musubi_Trainer_v8/Qwen_Image_Training_Models/qwen_train_vae.safetensors"
vae_dtype = "bfloat16"
vae_tiling = false
wandb_run_name = ""
weighting_scheme = "mode"
xformers = false
```

**Comment by FurkanGozukara** [comment]
@kohya-ss another issue is

when doing full BF16 training with mixed precision BF16, Flash Attention 2.8.3 loss rate is NaN - first step is accurate but then it becomes NaN

same settings SDPA works accurately

**Comment by kohya-ss** [comment]
The model parameters for FLUX.1 are 12B, while for Qwen-Image they are 20B, so in float32, the model weights alone will consume 80GB of RAM (20B * 4 bytes per parameter). If we take into account other modules, you will need more than 100GB of main RAM.

Also, due to differences in the block swap mechanism, if we do not apply the mechanism in PR #585 (i.e., in main), more than 80GB of shared memory will be required, meaning more than 160GB of main memory will be required.

Similarly, VRAM usage will increase with float32 precision. However, the amount of VRAM used is more than double when full_bf16 is specified and when it is not, so we will continue to investigate this.

The reason why the loss becomes NaN when flash attention is specified with full_bf16 may be due to differences in mathematical stability.

Edit: bf16 is recommended for mixed precision if you use bf16 weights (qwen_image_bf16).

**Comment by FurkanGozukara** [comment]
@kohya-ss thanks a lot
